webapp/packages/plugin-authentication-administration/src/Administration/Users/UsersTable/UserCredentialsList.tsx

```diff
@@ -32,11 +36,43 @@ export const UserCredentials = observer<IUserCredentialsProps>(function UserCred
 });
 
 export const UserCredentialsList: PlaceholderComponent<IUserDetailsInfoProps> = observer(function UserCredentialsList({ user }) {
-  return (
+  const styles = useS(style);
+  const translate = useTranslate();
+  const menu = useMenuState({
+    placement: 'top',
+    gutter: 8,
+  });
+
+  const visibleCredentials = user.origins.slice(0, MAX_VISIBLE_CREDENTIALS);
+
+  return styled(BASE_DROPDOWN_STYLES)(
     <Fragment key="user-credentials-list">
-      {user.origins.map(origin => (
+      {visibleCredentials.map(origin => (
         <UserCredentials key={origin.type + origin.subType} origin={origin} />
```

Comment:
string concatenation can sometimes lead to unexpected results. it is better to use `concat` method. In this case subtype can be undefined so it will lead to something like: **"originundefined"**. And if those keys are duplicated react can rerender incorrect

I believe it is better to give a unique id here with some `generateId` helper